### PR TITLE
Refactor: Radio Buttons

### PR DESCRIPTION
**Pull request by nemaniarjun**
There are 2 commits, one which refactors the use of the styled radio button on the leaderboards and submissions pages, and the other which creates a nice partial for the radio button and changes all checkboxes in the challenge edit page to use the new partial.

### CODE REVIEW

File: app/controllers/leaderboards_controller.rb [hunk 1]
```diff
@@ -6,13 +6,8 @@ class LeaderboardsController < ApplicationController
...
   def index
     @current_round = current_round
-    if @challenge.completed?
-      @post_challenge = if params[:post_challenge] == 'on'
-                          'on'
-                        else
-                          'off'
-                        end
-    end
+    @post_challenge = true if @challenge.completed? && params[:post_challenge].presence == "true"
```
Comment (piotrekpasciak) [inline app/controllers/leaderboards_controller.rb]: `params[:post_challenge].presence` - `presence` method is redundant here. It doesn't have any impact on this logical condition.

File: app/views/leaderboards/index.html.erb [hunk 2]
```diff
@@ -42,44 +42,21 @@
...
-				      <% end %>
-			      </div>
-		      </div>
-	      <% end %>
-      <!-- /toggle switch -->
+        <!-- toggle switch -->
+        <% if @challenge.status == :completed %>
+          <div class="toggle-switch-container">
+            <div class="can-toggle can-toggle--size-small">
+              <%= link_to challenge_leaderboards_path(challenge_round_id: @current_round, post_challenge: !@post_challenge) do %>
```
Comment (piotrekpasciak) [inline app/views/leaderboards/index.html.erb]: What's the purpose of `!@post_challenge`? Why do you have `!` operator here? Did you mean something like `@post_challenge.blank?`

File: app/views/submissions/index.html.erb [hunk 3]
```diff
@@ -49,31 +49,32 @@
...
-          <%= render partial: 'submissions/index/baselines_off',
-                locals: { challenge: @challenge, current_round_id: @current_round_id } %>
-          <% end %>
-          <% if @baselines == 'on' %>
-          <%= render partial: 'submissions/index/baselines_on',
-                locals: { challenge: @challenge, current_round_id: @current_round_id } %>
-          <% end %>
+          <div class="toggle-switch-container">
+            <div class="can-toggle can-toggle--size-small">
+              <%= link_to challenge_submissions_path(@challenge, my_submissions: !@my_submissions, current_round_id: @current_round_id) do %>
```
Comment (piotrekpasciak) [inline app/views/submissions/index.html.erb]: `!@my_submissions` - same questions like above.

File: app/views/submissions/index.html.erb [hunk 4]
```diff
@@ -49,31 +49,32 @@
...
+                <label for="my-submissions">
+                  <div class="can-toggle__switch" data-checked="On" data-unchecked="Off"></div>
+                  <div class="can-toggle__label-text">My Submissions</div>
+                </label>
+              <% end %>
+            </div>
+          </div>
+          <div class="toggle-switch-container">
+            <div class="can-toggle can-toggle--size-small">
+              <%= link_to challenge_submissions_path(@challenge, baselines: !@baselines, current_round_id: @current_round_id) do %>
```
Comment (piotrekpasciak) [inline app/views/submissions/index.html.erb]: `!baselines`

File: app/views/leaderboards/index.html.erb (hunk 2 above)
Comment (nemaniarjun) [reply app/views/leaderboards/index.html.erb]: if @post_challenge is `nil` we want to pass `true` here if its `true` we want to pass `false` here

File: app/controllers/leaderboards_controller.rb (hunk 1 above)
Comment (nemaniarjun) [reply app/controllers/leaderboards_controller.rb]: aah, will change this!

File: app/views/leaderboards/index.html.erb (hunk 2 above)
Comment (piotrekpasciak) [reply app/views/leaderboards/index.html.erb]: Can we use `@post_challenge.blank?` in that case? It's hard to read code with `!` operators.

Comment (nemaniarjun) [reply app/views/leaderboards/index.html.erb]: initally I was setting @post_challenge to `false` that is why I was using `!` but now in the controller we are only setting it if its true so yes I will change it to `.blank` that is definitely more readable

**Pull request closed without merging**